corehq/apps/accounting/views.py

```diff
@@ -927,6 +931,88 @@ def page_context(self):
         return context
 
 
+class CustomerInvoiceSummaryView(InvoiceSummaryViewBase):
+    urlname = 'customer_invoice_summary'
+    invoice_class = CustomerInvoice
+
+    @property
+    def parent_pages(self):
+        return [{
+            'title': CustomerInvoiceInterface.name,
+            'url': CustomerInvoiceInterface.get_url()
+        }]
+
+    @property
+    @memoized
+    def adjust_balance_form(self):
+        if self.request.method == 'POST':
+            return AdjustBalanceForm(self.invoice, self.request.POST)
+        return AdjustBalanceForm(self.invoice)
+
+    @property
+    @memoized
+    def billing_records(self):
+        return self.invoice.customerbillingrecord_set.all()
+
+    @property
+    @memoized
+    def adjustment_list(self):
+        adjustment_list = CreditAdjustment.objects.filter(customer_invoice=self.invoice)
+        return adjustment_list.order_by('-date_created')
+
+    @property
+    def can_send_email(self):
+        return True
+
+    @property
+    def page_context(self):
+        context = super(CustomerInvoiceSummaryView, self).page_context
+        context.update({
+            'adjustment_balance_form': self.adjust_balance_form,
+            'adjustment_list': self.adjustment_list
+        })
+        return context
+
+
+class InvoicePdfView(View):
```

Comment:
GitHub won't show me the line of code you're referencing, but I think you're talking about the `CustomerInvoiceSummaryView` class, which is the page you land on when you view an invoice (not its PDF)
<img width="1376" alt="screen shot 2018-07-06 at 1 35 31 pm" src="https://user-images.githubusercontent.com/15271571/42392667-984daf2e-8121-11e8-8a7b-472db0e06561.png">

Reply:
Sorry, I meant ```InvoicePdfView```.  I saw you defined this class and included it in the urls.py, but didn't see the url being used.

Reply:
Ah, that makes so much more sense!

It's used to view PDFs generated from customer-level invoices, specifically (after going through `CustomerInvoiceInterface` and `CustomerInvoiceSummaryView` you're going to get to `InvoicePdfView `).

The `customer_invoices` slug is in `invoice.html`:
```
{% if billing_record.invoice.is_customer_invoice %}
                                <td>
                                    <a href="{% url 'invoice_pdf_view' billing_record.pdf_data_id %}">
                                        View PDF
                                    </a>
                                </td>
                            {% else %}
                                <td>
                                    <a href="{% url 'domain_billing_statement_download' billing_record.invoice.get_domain billing_record.pdf_data_id %}">
                                        View PDF
                                    </a>
                                </td>
                            {% endif %}
```

Reply:
👍

Reply:
small nit: it might be helpful for future devs to rename ```InvoicePdfView``` to ```CustomerInvoicePdfView``` or similar to clarify that this is not intended for subscription level invoices (not blocking merging)